### PR TITLE
[cli] Add `sanity init --bare` flag

### PR DESCRIPTION
**Pull request by rexxars**
**Type of change (check at least one)**

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Documentation (fix or update to documentation)
- [ ]  Maintenance
- [ ]  Other, please describe:

**Does this change require a documentation update? (Check one)**

- [x]  Yes
- [ ]  No

**Description**

This PR allows a user to run `sanity init --bare` and only create a project and (optionally) a dataset. If the user chooses to, it will also create a folder with a `sanity.json` (containing project ID and dataset) and a `package.json` (containing `@sanity/core` as a dependency).

I didn't want to create even more "branches" in the init project code which is already a bit unwieldy, so I separated some logic into their own files for reuse between the "init bare" method and the "regular" init.

**Note for release**

Add `--bare` flag to CLI `sanity init` command, for cases where a studio is not needed.

**Checklist** 

- [x]  I have read the [Contributing Guidelines](https://github.com/sanity-io/sanity/blob/next/CONTRIBUTING.md)
- [ ]  The PR title includes a link to the relevant issue
- [x]  The PR title is appropriately formatted: `[some-package] PR title (#123)`
- [x]  The code is linted
- [x]  The test suite is passing
- [ ]  Corresponding changes to the documentation have been made

### CODE REVIEW

File: packages/@sanity/cli/src/commands/init/initCommand.js [hunk 1]
```diff
@@ -12,6 +13,7 @@ Options
   --visibility <mode> Visibility mode for dataset (public/private)
   --create-project <name> Create a new project with the given name
   --reconfigure Reconfigure Sanity studio in current folder with new project/dataset
+  --bare Do not create any studio files, only sanity.json/package.json
```
Comment (kmelve) [inline packages/@sanity/cli/src/commands/init/initCommand.js:16]:
```suggestion
  --bare Create a project without a studio, only sanity.json and package.json
```

File: packages/@sanity/cli/src/commands/init/initCommand.js [hunk 2]
```diff
@@ -30,6 +32,9 @@ Examples
   # template to the given path
   sanity init -y --project abc123 --dataset staging --template moviedb --output-path .
 
+  # Initialize a "bare" project, only creating sanity.json and package.json
```
Comment (kmelve) [inline packages/@sanity/cli/src/commands/init/initCommand.js:35]:
```suggestion
  # Initialize a project without Sanity Studio, only creating sanity.json and package.json
```

**Review by kmelve — CHANGES_REQUESTED** [review]
This is awesome! A couple of suggestions for wordings only

**Review by kmelve — APPROVED** [review]

**Review by vicbergquist — APPROVED** [review]
LGTM!

**Comment by rexxars** [comment]
Will revisit this later.

**Pull request closed without merging**